datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -478,7 +478,13 @@ public ActionResult<Void> stop(@PathKeysParam PathKeys pathKeys,
           // this check helps in preventing any datastream from being stuck in STOPPING state indefinitely
           LOG.warn("Datastream {} is already in {} state. Notifying leader to initiate transition", d,
               d.getStatus());
-          _store.updateDatastream(d.getName(), d, true);
+          if (force) {
```

Comment:
1. Logs should reflect that it was a force stop.
2. Add comments explaining the rationale.

Reply:
Done